tests/asgi/_asgi_test_app.py

```diff
@@ -108,11 +108,15 @@ async def on_post(self, req, resp):
         resp.text = await req.stream.read()
 
     async def on_put_drops(self, req, resp):
+        # NOTE(kgriffs): Integrity check
+        sha1 = hashlib.sha1()
```

Comment:
let's use something different, even if not necessary since static analysis tools with their immense wisdom may flag it as "unsecure algorithm used" or something similar.

python 3.5 has sha256, 384 and 512, so we could use one of those

Reply:
For integrity checking even CRC32 is still very prevalent, and you surely use it every day, both for your TCP network packets and your disk's sectors, and it doesn't mean it is "insecure". Furthermore, we already use `sha1` for other tests, so we should make a consistent change throughout the whole codebase if we decide this.
So make we could switch to [`crc32()`](https://docs.python.org/3.10/library/zlib.html#zlib.crc32) to make it obvious that it is merely a checksum? :smiling_imp:

@CaselIT do you know of any such tools that would flag references to `sha1`?

Reply:
I know that here it's not an issue..

> @CaselIT do you know any such tools that would flag references to `sha1`?

sonarqube will flag it, but I don't seem to find a list of rules. I'm sure it's not the only one.

Note that I think we can merge as is, just mentioned since static analysis tools can be quite dumb at times

Reply:
Yeah, that's a good point regarding static analysis tools. I was genuinely curious, good to know wrt sonarcube.

OTOH I don't think we should care too much about these because is inside the `tests/` directory, which is not installed when installing `falcon`. So it is not an attack vector even if it was a real issue.

I'm not opposed to changing to another hashing function, but if we decide to change, it makes sense to update all the tests to use our new checksum function of choice.

Reply:
makes sense. let's keep the change in mind, and if we decide to do it we should change them in a single issue.